WordPress/Sniffs/WP/DeprecatedParametersSniff.php

```diff
@@ -0,0 +1,352 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Check for usage of deprecated parameters in WP functions and sugests alternative based on the parameter passed.
+ *
+ * This sniff will throw an error when usage of deprecated parameters is
+ * detected if the parameter was deprecated before the minimum supported
+ * WP version; a warning otherwise.
+ * By default, it is set to presume that a project will support the current
+ * WP version and up to three releases before.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.11.0
+ */
+class WordPress_Sniffs_WP_DeprecatedParametersSniff extends WordPress_AbstractFunctionParameterSniff {
+
+	/**
+	 * The group name for this group of functions.
+	 *
+	 * @since 0.11.0
+	 *
+	 * @var string
+	 */
+	protected $group_name = 'wp_deprecated_parameters';
+
+	/**
+	 * Minimum WordPress version.
+	 *
+	 * This variable allows changing the minimum supported WP version used by
+	 * this sniff by setting a property in a custom ruleset xml file.
+	 *
+	 * Example usage:
+	 * <rule ref="WordPress.WP.DeprecatedParameters">
+	 *  <properties>
+	 *   <property name="minimum_supported_version" value="4.5"/>
+	 *  </properties>
+	 * </rule>
+	 *
+	 * @since 0.11.0
+	 *
+	 * @var string WordPress version.
+	 */
+	public $minimum_supported_version = 4.5;
+
+	/**
+	 * Array of function, argument, and default value for deprecated argument.
+	 *
+	 * The functions are ordered by the WordPress version the parameter was
+	 * deprecated in. Was last updated in WordPress version 4.8.0.
+	 *
+	 * @since 0.12.0
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *
+	 *               @type string Function name. {
+	 *                   @type int target Parameter positions. {
+	 *                       @type string|boolean|null  Default value.
```

Comment:
> (mixed) Default value. Currently boolean, null, array and strings are supported. Support for additional types can be added to $this->process_parameters().

I'd suggest breaking this long comment up into several lines and maybe making it a little more specific. Something along the lines of:
> (mixed) Expected default value for the deprecated parameter.
> Currently the default values: `true`, `false`, `null`, empty arrays and both empty and non-empty strings can be handled correctly by the `process_parameters()` method.
> If/when an additional default value would be added, the relevant code in the `process_parameters()` method will need to be adjusted.

Note: this phrasing can be definitely still be improved - @GaryJones suggestions ?